analyzer/src/main/kotlin/managers/Yarn.kt

```diff
@@ -69,6 +70,12 @@ class Yarn(
 
     override fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode {
         val process = run(workingDir, "info", "--json", packageName)
-        return jsonMapper.readTree(process.stdout)["data"]
+        val yarnInfo = jsonMapper.readTree(process.stdout)["data"]
+        if (yarnInfo == null) {
+            val yarnErr = jsonMapper.readTree(process.stderr)["data"]
+            log.warn("Error running `yarn info --json $packageName` in directory $workingDir: $yarnErr")
+            return yarnErr
+        }
+        return yarnInfo
```

Comment:
You can write this a bit more compact:
```kotlin
return jsonMapper.readTree(process.stdout)["data"] ?: jsonMapper.readTree(process.stderr)["data"].also {
  log.warn("Error running `yarn info --json $packageName` in directory $workingDir: $yarnErr")
}
```

And does the error message make sense? Does `yarnErr` contain the actual error message? Also you should use `process.commandLine` instead of manually repeating the command line in the message.

Regarding the commit message, is the VCS information really the only thing missing in case of the fallback?

Reply:
And in any case please the curly-brace syntax for log statements: `log.warn { ... }`